### PR TITLE
Add mcs shared interface and controller

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Add mcs shared interface and controller to sync MultiClusterService.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add MultiClusterService shared interface and multiclusterservice-controller
```

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
/assign

File: app/controlers.go [hunk 1]
```diff
@@ -32,6 +33,28 @@ func startMCIController(ctx controllerscontext.Context) (enabled bool, err error
 	return true, nil
 }
 
+func startMCSController(ctx controllerscontext.Context) (enabled bool, err error) {
+	loadBalancer, support := ctx.CloudProvider.MCSLoadBalancer()
+	if !support {
+		return false, fmt.Errorf("the multicluster controller manager does not support external loadBalancer")
+	}
+	if loadBalancer == nil {
+		return false, fmt.Errorf("clould get the target external loadBalancer provider")
```
Comment (RainbowMango) [inline app/controlers.go]: typo? `clould`

Comment (XiShanYongYe-Chang) [reply app/controlers.go]: It's a typo indeed, and I'll modify it

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**